File: src/main/java/net/dontdrinkandroot/gitki/config/GitkiConfiguration.java
package net.dontdrinkandroot.gitki.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
@Configuration
public class GitkiConfiguration
{
    //    @Bean
    //    @ConditionalOnProperty("gitki.oauth.github.enabled")
    //    @ConfigurationProperties("gitki.oauth.github.client")
    //    public AuthorizationCodeResourceDetails github()
    //    {
    //        return new AuthorizationCodeResourceDetails();
    //    }
    //
    //    @Bean
    //    @ConditionalOnProperty("gitki.oauth.github.enabled")
    //    @ConfigurationProperties("gitki.oauth.github.resource")
    //    public ResourceServerProperties githubResource()
    //    {
    //        return new ResourceServerProperties();
    //    }

    @Bean
    public PasswordEncoder passwordEncoder()
    {
        return new BCryptPasswordEncoder();
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/config/GitkiConfiguration.kt
package net.dontdrinkandroot.gitki.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder

@Configuration
class GitkiConfiguration {

    //    @Bean
    //    @ConditionalOnProperty("gitki.oauth.github.enabled")
    //    @ConfigurationProperties("gitki.oauth.github.client")
    //    public AuthorizationCodeResourceDetails github()
    //    {
    //        return new AuthorizationCodeResourceDetails();
    //    }
    //
    //    @Bean
    //    @ConditionalOnProperty("gitki.oauth.github.enabled")
    //    @ConfigurationProperties("gitki.oauth.github.resource")
    //    public ResourceServerProperties githubResource()
    //    {
    //        return new ResourceServerProperties();
    //    }
    @Bean
    fun passwordEncoder(): PasswordEncoder = BCryptPasswordEncoder()
}